fuse_core/include/fuse_core/message_buffer_impl.h

```diff
@@ -131,17 +131,17 @@ template<class Message>
 void MessageBuffer<Message>::purgeHistory()
 {
   // Purge any messages that are more than buffer_length_ seconds older than the most recent entry
-  // A setting of ros::DURATION_MAX means "keep everything"
+  // A setting of rclcpp::Duration::max() means "keep everything"
   // And we want to keep at least two entries in buffer at all times, regardless of the stamps.
-  if ((buffer_length_ == ros::DURATION_MAX) || (buffer_.size() <= 2))
+  if ((buffer_length_ == rclcpp::Duration::max()) || (buffer_.size() <= 2))
   {
     return;
   }
 
   // Compute the expiration time carefully, as ROS can't handle negative times
   const auto& ending_stamp = buffer_.back().first;
   auto expiration_time =
-      ending_stamp.toSec() > buffer_length_.toSec() ? ending_stamp - buffer_length_ : ros::Time(0, 0);
+      ending_stamp.toSec() > buffer_length_.toSec() ? ending_stamp - buffer_length_ : rclcpp::Time(0, 0);
```

Comment:
`rclcpp::Time(0,0)` is tricky because [the default clock with this constructor is system time](https://github.com/ros2/rclcpp/blob/3aca271ef5d25c1b8c12e67ecf85166856a88000/rclcpp/include/rclcpp/time.hpp#L46), and that would cause a comparison failure if a different clock was used in a time in the `buffer_`. I'd recommend `rclcpp::Time(0, 0, ending_stamp.get_clock_type())` instead so the comparisons will work assuming all times in the buffer have the same clock type.

Reply:
The only places I've found time zero used are when:
* the logic is actually testing for uninitialised variables, not time values
* the logic has set a time value to an invalid state to force the result of a conditional statement (this example)
* both

`StampedVariableSynchronizer` uses it quite a bit and gets twisty as a result.
It might make sense to try patching out the use of uninitialised time in the ROS1 version while we figure out the rest of the requirements for a fuse_core time class